utils_test.go

```diff
@@ -97,6 +98,41 @@ func (r With) Path(p string) string {
 	return fmt.Sprintf("%s%s", r, p)
 }
 
+func HandleWs(w http.ResponseWriter, r *http.Request) {
+	upgrader := websocket.Upgrader{
+		CheckOrigin: func(_ *http.Request) bool {
+			return true
+		},
+	}
+	c, err := upgrader.Upgrade(w, r, nil)
+	if err != nil {
+		panic(err)
+	}
+	defer c.Close()
+	for {
+		mt, message, err := c.ReadMessage()
+		if err != nil {
+			break
+		}
+		type req struct {
+			ID uint64 `json:"id"`
+		}
+		var r req
+		err = json.Unmarshal(message, &r)
+		if err != nil {
+			panic(err)
+		}
+		output, err := json.Marshal(r)
```

Comment:
Here you are just checking "r is a valid json"?

Reply:
Yes